gpflow/multioutput/conditionals.py

```diff
@@ -120,12 +120,12 @@ def single_gp_conditional(t):
                           (Kmms, Kmns, Knns, fs, q_sqrts),
                           (settings.float_type, settings.float_type))  # P x N x 1, P x 1 x N x N or P x N x 1
 
-    fmu = tf.matrix_transpose(rmu[:, :, 0])  # N x P
+    fmu = _rollaxis(rmu[..., 0], 1)  # N x P
 
     if full_cov:
-        fvar = rvar[:, 0, :, :]  # P x N x N
+        fvar = rvar[..., 0, :, :]  # P x N x N
```

Comment:
why does it have a dimension of length 1 in it anyways ?

Reply:
This is due to the `tf.map` used above